app/forms/ImporterEoriFormProvider.scala

```diff
@@ -26,6 +26,10 @@ class ImporterEoriFormProvider @Inject() extends Mappings {
   def apply(): Form[EORI] =
     Form(
       "value" -> text("importerEori.error.required")
-        .verifying(maxLength(17, "importerEori.error.length")).transform[EORI](EORI.apply, _.value)
+        .verifying(firstError(
+          maxLength(17, "importerEori.error.length"),
```

Comment:
@somaraja if the error message is the same for but of these validation checks, then I don't think we need the maxLength check as the regexp check will check the length too.